Index: src/event/ngx_event_openssl.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/event/ngx_event_openssl.h b/src/event/ngx_event_openssl.h
--- a/src/event/ngx_event_openssl.h	(revision 4acdfe3f86599ada0bfcca95bb7950f1c5b64c1d)
+++ b/src/event/ngx_event_openssl.h	(revision 108e3850ce6156bee9c2f89c1eae600edea3b245)
@@ -128,6 +128,25 @@
     unsigned                    in_ocsp:1;
     unsigned                    early_preread:1;
     unsigned                    write_blocked:1;
+
+    // ja4
+    int             version;
+    int             highest_supported_tls_client_version;
+
+    size_t          ciphers_sz;
+    char **ciphers;
+
+    size_t          extensions_sz;
+    char **extensions;
+
+    size_t          sigalgs_sz;
+    char   **sigalgs_hash_values; // Array to store combined hash values like 0x0601
+    char *first_alpn; // first ALPN protocol provided by the client
+
+
+    // ja4l
+    uint16_t handshake_roundtrip_microseconds; // a whole number - max is probably thousands
+    uint8_t ttl; // time to live - a whole number - max is 255
 };
 
 
